sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_internal/async_challenge_auth_policy.py

```diff
@@ -69,6 +71,15 @@ async def on_challenge(self, request: "PipelineRequest", response: "PipelineResp
         except ValueError:
             return False
 
+        if self._verify_challenge_resource:
+            resource_domain = urlparse(scope).netloc
+            if not resource_domain:
+                raise ValueError(f"The challenge contains invalid scope '{scope}'.")
+
+            request_domain = urlparse(request.http_request.url).netloc
+            if not request_domain.lower().endswith(f".{resource_domain.lower()}"):
```

Comment:
We don't need to do any further stipping of "/" characters or anything?
I'm pretty sure `netloc` shouldn't include these.... what about things like port numbers? Will that ever interfere? I can't recall exactly what formatting `netloc` returns.

Reply:
That's a good point. I'll see if I can add a test to specifically check for port numbers; our auth policy is currently port-agnostic when considering the endpoint of a request, but I need to make sure this particular check wouldn't be disrupted